torch/onnx/_internal/fx/fx_onnx_interpreter.py

```diff
@@ -128,15 +128,41 @@ def _retrieve_or_adapt_input_to_graph_set(
                 List[int],
             ]
         ] = []
+        # onnx_tensor contains a list of scalars which could be one of
+        #   - tensor with empty shape,
+        #   - tensor with tensor with shape (1,),
+        #   - torch.SymInt,
+        #   - int
+        #   - ...
+        # They should all be promoted to tensor with shape (1,)
+        # in order to call ONNX's Concat.
         for tensor in onnx_tensor:
+            # Prepare `tensor` as input of ONNX's Concat.
+
             if isinstance(
                 tensor, torch.fx.Node
             ) and fx_type_utils.is_torch_symbolic_type(tensor.meta.get("val")):
-                sequence_mixed_elements.append(fx_name_to_onnxscript_value[tensor.name])
+                # In this case, tensor is a torch.SymInt from Dynamo's perspective.
+                # It might be mapped to tensor with shape () or (1,) in ONNX.
+                element_value = fx_name_to_onnxscript_value[tensor.name]
```

Comment:
I share a similar concern as @BowenBao. The comments make clear what is being done, but it is not clear to me whether this is an unintended behavior for other models. That is, is it really ok to always promote a scalar to tensor here?

Reply:
Some if-else conditions before this statement are listed below. They imply an input is a list of `Union[SymInt, int, ...]` in FX. To make them a tensor in ONNX, the only way is to concat them. To concat them, you first need to make them all (1,)-tensor since scalars cannot be concated. This has been implemented below for `int`. Here we just implement the same run for Sym*.
```            if isinstance(
                tensor, torch.fx.Node
            ) and fx_type_utils.is_torch_symbolic_type(tensor.meta.get("val")):
                # In this case, tensor is a torch.SymInt from Dynamo's perspective.
                # It might be mapped to tensor with shape () or (1,) in ONNX.
                element_value = fx_name_to_onnxscript_value[tensor.name]
                if isinstance(
                    element_value, onnxscript_graph_building.TorchScriptTensor
                ):
 ```